llvm/tools/sycl-post-link/sycl-post-link.cpp

```diff
@@ -353,6 +353,18 @@ std::string saveModuleIR(Module &M, int I, StringRef Suff) {
   return OutFilename;
 }
 
+static constexpr const char *REG_ALLOC_MODE_ATTR_NAME = "RegisterAllocMode";
+
+static bool getRegAllocModeValue(const Function &F, uint32_t &Val) {
+  if (!F.hasFnAttribute(REG_ALLOC_MODE_ATTR_NAME))
+    return false;
+  StringRef RegAllocModeValAsStr =
+      F.getFnAttribute(REG_ALLOC_MODE_ATTR_NAME).getValueAsString();
+  if (!llvm::to_integer(RegAllocModeValAsStr, Val))
+    assert(false && "Invalid metadata");
```

Comment:
You can use [`StringRef::getAsInteger`](https://llvm.org/doxygen/classllvm_1_1StringRef.html#a1881146f2dcc2ca57c9c5f77f938db9d) here to reduce amount of different APIs (and potentially missed `#include`s)

Reply:
nice, thanks